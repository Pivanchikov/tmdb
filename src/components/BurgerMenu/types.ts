export type MenuProps = {
  items: {
    top: string[]
    middle: string[]
    bottom: string[]
  }
}
