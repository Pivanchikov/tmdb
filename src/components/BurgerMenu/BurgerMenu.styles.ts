export const styles = theme => ({
  root: {
    backgroundColor: theme.palette.primary.main,
    marginTop: 64,
    minWidth: 250,
    paddingLeft: 5,
  },
  list: {
    fontSize: '1.3rem',
    fontWeight: 600,
    color: theme.palette.secondary.main,
  },
  icon: {
    color: theme.palette.secondary.main,
  },
  item: {
    color: theme.palette.secondary.main,
  },
})
