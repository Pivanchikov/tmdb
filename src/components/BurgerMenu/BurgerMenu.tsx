import { useState } from 'react'
import { IconButton, List, makeStyles, SwipeableDrawer, ListItem, ListItemText } from '@material-ui/core'
import MenuTwoToneIcon from '@material-ui/icons/MenuTwoTone'
import { useTranslation } from 'react-i18next'

import { styles } from './BurgerMenu.styles'
import { MenuProps } from './types'

const useStyles = makeStyles(styles)

export const BurgerMenu = ({ items }: MenuProps) => {
  const classes = useStyles()
  const [opened, setOpened] = useState(false)
  const { t } = useTranslation()

  const toggleDrawer = () => setOpened(state => !state)

  return (
    <>
      <IconButton onClick={toggleDrawer} className={classes.icon}>
        <MenuTwoToneIcon fontSize="large" />
      </IconButton>
      <SwipeableDrawer
        anchor={'left'}
        open={opened}
        onClose={toggleDrawer}
        onOpen={toggleDrawer}
        classes={{ paper: classes.root }}
      >
        <List>
          {items.top.map(text => (
            <ListItem button key={text}>
              <ListItemText primary={t(text)} classes={{ primary: classes.list }} />
            </ListItem>
          ))}
        </List>
        <List>
          {items.middle.map(text => (
            <ListItem button key={text}>
              <ListItemText primary={t(text)} classes={{ primary: classes.item }} />
            </ListItem>
          ))}
        </List>
        <List>
          {items.bottom.map(text => (
            <ListItem button key={text}>
              <ListItemText primary={t(text)} classes={{ primary: classes.item }} />
            </ListItem>
          ))}
        </List>
      </SwipeableDrawer>
    </>
  )
}
