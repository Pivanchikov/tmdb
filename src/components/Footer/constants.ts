export const linkName = ['about', 'faq']
export const copyright = 'Â© 2021 TMDb, inc.'
