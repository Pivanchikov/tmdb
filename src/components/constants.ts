export const films = ['popular', 'nowPlaying', 'upcoming', 'topRated']
export const serials = ['popular', 'airingToday', 'onTv', 'topRated']
export const people = ['popularPeople']
export const another = ['support', 'API']
export const burgerItems = {
  top: ['movies', 'tvShows', 'people'],
  middle: ['API', 'support', 'about'],
  bottom: ['signIn', 'signUp'],
}
export const languages = [
  ['Ru', 'ru'],
  ['Gb', 'en'],
]
export const desktopWidth = 1280
export const TIMEOUT_MS = 300
export const QUERY_LENGTH_THRESHOLD = 3
