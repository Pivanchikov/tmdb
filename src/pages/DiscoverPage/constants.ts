export const STEP = 1
export const MIN_YEAR = 1920
export const MAX_YEAR = 2021
