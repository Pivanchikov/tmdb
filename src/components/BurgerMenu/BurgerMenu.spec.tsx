import { render as testingLibraryRender, fireEvent } from '@testing-library/react'

import { BurgerMenu } from './BurgerMenu'
import { burgerItems } from '../constants'

jest.mock('react-i18next', () => ({
  useTranslation: () => ({
    t: jest.fn(value => value),
    i18n: {
      language: 'en',
      changeLanguage: jest.fn(),
    },
  }),
}))

describe('BurgerMenu', () => {
  const render = () => testingLibraryRender(<BurgerMenu items={burgerItems} />)

  it('renders correctly initial document', () => {
    const { getByRole } = render()

    expect(getByRole('button')).toBeInTheDocument
  })

  it('click on menu button', () => {
    const { getAllByRole, getByRole, getByText } = render()
    const expectedLength = 8

    fireEvent.click(getByRole('button'))

    expect(getAllByRole('button')).toHaveLength(expectedLength)
    burgerItems.top.forEach(item => expect(getByText(item)).toBeInTheDocument)
    burgerItems.middle.forEach(item => expect(getByText(item)).toBeInTheDocument)
    burgerItems.bottom.forEach(item => expect(getByText(item)).toBeInTheDocument)
  })

  it('matches snapshot', () => {
    const { asFragment } = render()
    expect(asFragment()).toMatchSnapshot()
  })
})
