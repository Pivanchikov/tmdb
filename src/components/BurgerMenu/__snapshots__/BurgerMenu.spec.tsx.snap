// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BurgerMenu matches snapshot 1`] = `
<DocumentFragment>
  <button
    class="MuiButtonBase-root MuiIconButton-root makeStyles-icon-21"
    tabindex="0"
    type="button"
  >
    <span
      class="MuiIconButton-label"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
        />
      </svg>
    </span>
    <span
      class="MuiTouchRipple-root"
    />
  </button>
  <div
    class="PrivateSwipeArea-root-23 PrivateSwipeArea-anchorLeft-24"
    style="width: 20px;"
  />
</DocumentFragment>
`;
